docs/specification.md

```diff
@@ -370,30 +373,112 @@ when verification was in progress locally. If this is the case, the provisional
+
+This mechanism is utilized in scenarios where some change is rejected. For
+example, one or more commits may have been pushed to a branch that do not meet
+gittuf policy. The repository is updated such that these commits are neutralized
+and all Git refs match their latest RSL entries. This can take two forms:
+
+1. The rejected commit is removed and the state of the repo is set to the prior
+commit which is known to be good. This is used when all rejected commits are
+together at the end of the commit graph, making it easy to remove all of them.
+
+2. The rejected commit is _reverted_ where a new commit is introduced that
+reverses all the changes made in the reverted commit. This is needed when "good"
+commits that must be retained are interspersed with "bad" commits that must be
+rejected.
+
+In both cases, new RSL entries and annotations may be used to record the
+incident or to skip the invalid RSL entries corresponding to the rejected
+changes. TODO: should these new RSL entries come from authorized users for the
+affected namespace.
+
+#### M2: Create RSL entry on behalf of another user
+
+This mechanism is necessary for adoptions where a subset of developers do not
+use gittuf. When they submit changes to the main copy of the repository, they do
+not include RSL entries. Therefore, when a change is pushed to a branch by a
+non-gittuf user A, a gittuf user B can submit an RSL entry on their behalf.
+Additionally, the entry must identify the original user (TODO: cross-reference
+with signed pushes?) and include some evidence about why B thinks the change
+came from A (TODO: signed push can be evidence option?).
+
+TODO: decide if B needs to be authorized for the affected branch to sign for A.
+
+### Recovery Scenarios
+
+These scenarios are some examples where recovery is necessary.
+
+#### A change is made without an RSL entry
+
+Bob does not use gittuf and pushes to a branch. Alice notices this as her gittuf
+client detects a push to the branch without an accompanying RSL entry. She
+validates that the change came from Bob and creates an RSL entry on his behalf,
+identifying him and including information about how she verified it was him.
+Alice applies M2.
+
+#### An incorrect RSL entry is added
+
+If an entry is malformed (structurally), Git should catch it if it's not a valid
+commit. In addition, gittuf must catch any valid commits with malformed RSL
+data. For example, a necessary field may be missing, or may contain invalid
+data. gittuf must reject such entries and warn the user when they are
+encountered. As this scenario requires the author's client to be buggy or
+malicious, it cannot be trusted to catch the invalid entries when the actor
+pushes to the main copy of the repository. However, correct gittuf clients
+verify that RSL entries are not invalid when they pull from the main copy of
+the repository. If the main copy is enabled with gittuf, such incidents can be
+caught before other users receive the invalid changes.
```

Comment:
Is there any type of invalid RSL entry which a gittuf client will not be able to detect upon pull?

Reply:
I can't think of any at the moment. RSL entries don't contain a ton of information so we should be able to verify the validity of whatever they contain...

Reply:
Okay, be sure to ensure that things like "the actual commit pointed to is sensible" in addition to "the hash is a hash and points to a commit"..

Reply:
Enumerated some invalid entry examples.